### PR TITLE
Flat is Justice

### PR DESCRIPTION
**Pull request by Nekotekina**
https://en.wikipedia.org/wiki/Flat_memory_model :blush:

### CODE REVIEW

File: rpcs3/Emu/RSX/VK/VKTextureCache.h [hunk 1]
```diff
@@ -309,7 +309,7 @@ namespace vk
 			AUDIT( valid_length > 0 );
 
 			void* pixels_src = dma_buffer->map(valid_offset, valid_length);
-			void* pixels_dst = get_ptr_by_offset(valid_offset, true);
```
Comment (ruipin) [inline rpcs3/Emu/RSX/VK/VKTextureCache.h]: I believe this is incorrect.
`get_section_base()` returns the section base (`cpu_range.start`), but we want `cpu_range.start + valid_offset`.

Should be `get_ptr(get_section_base() + valid_offset)`, I think

File: rpcs3/Emu/RSX/GL/GLTextureCache.h [hunk 2]
```diff
@@ -495,7 +495,7 @@ namespace gl
 			const u32 valid_length = valid_range.second;
 			AUDIT( valid_length > 0 );
 
-			void *dst = get_ptr_by_offset(valid_range.first, true);
```
Comment (ruipin) [inline rpcs3/Emu/RSX/GL/GLTextureCache.h]: See other comment.

File: rpcs3/Emu/RSX/GL/GLTextureCache.h [hunk 3]
```diff
@@ -495,7 +495,7 @@ namespace gl
...
+			void *dst = get_ptr(get_section_base());
```
Comment (ruipin) [inline rpcs3/Emu/RSX/GL/GLTextureCache.h]: I believe this is incorrect.
`get_section_base()` returns the section base (`cpu_range.start`), but we want `cpu_range.start + valid_offset`.

Should be `get_ptr(get_section_base() + valid_offset)`, I think

File: rpcs3/Emu/RSX/VK/VKTextureCache.h [hunk 4]
```diff
@@ -309,7 +309,7 @@ namespace vk
...
+			void* pixels_dst = get_ptr(get_section_base());
```
Comment (ruipin) [inline rpcs3/Emu/RSX/VK/VKTextureCache.h]: I believe this is incorrect.
`get_section_base()` returns the section base (`cpu_range.start`), but we want `cpu_range.start + valid_offset`.

Should be `get_ptr(get_section_base() + valid_offset)`, I think

File: rpcs3/Emu/RSX/GL/GLTextureCache.h (hunk 3 above)
Comment (Nekotekina) [reply rpcs3/Emu/RSX/GL/GLTextureCache.h]: Ok, fixed.

**Comment by 0x8080** [comment]
As a fellow ~~lolicon~~ SW dev I agree with this

**Comment by mrtawsiti3** [comment]
i agree

**Comment by Daist** [comment]
This build (compared to others prior to 0.0.5-7405) improves framerate in TLOU menu (both BCES01585 and BCUS98174) from approx. 10 fps to 17-18 fps on 8700k@4.8GHz (both ASMJIT and LLVM). 
Ingame framerate with ASMJIT has not changed though (cannot check ingame with LLVM due to crashes).

**Comment by ruipin** [comment]
TLOU was actually the game where I first detected the `rsx::weak_ptr` inefficiencies that this PR removes, so that is to be expected.

**Pull request merged**